liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -180,6 +171,19 @@ private void generateFeatures() throws MojoExecutionException, PluginExecutionEx
             scannedFeatureList = binaryScannerHandler.runBinaryScanner(existingFeatures, classFiles, directories, eeVersion, mpVersion, optimize);
         } catch (BinaryScannerUtil.NoRecommendationException noRecommendation) {
             throw new MojoExecutionException(String.format(BinaryScannerUtil.BINARY_SCANNER_CONFLICT_MESSAGE3, noRecommendation.getConflicts()));
+        } catch (BinaryScannerUtil.FeatureModifiedException featuresModified) {
+            Set<String> userFeatures = getServerFeatures(servUtil, generatedFiles, true); // user features excludes generatedFiles
```

Comment:
If we want to avoid calling `getServerFeatures(..)` more than we have to, we could add a check here where if (optimize=true) then use the `existingFeatures` list above. 
Then the improvement may be pretty minimal so we can leave it as is, I do not have a strong preference either way.

Reply:
Thank you, that is a good idea because `getServerFeatures(..)` can be expensive.